constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintLayout.java

```diff
@@ -1642,6 +1641,7 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         if (DEBUG) {
             time = System.currentTimeMillis();
         }
+        mDirtyHierarchy |= dynamicUpdateConstraints( widthMeasureSpec,  heightMeasureSpec);
```

Comment:
Can this really modify the hierarchy? What sort of usage would trigger this?

Reply:
Bad choice of words. This flag is just saying if the layout needs remeasuring. Depending on what you change you may need remeasuring.